iceoryx_posh/include/iceoryx_posh/runtime/service_discovery.hpp

```diff
@@ -46,7 +46,17 @@ class ServiceDiscovery
                 const cxx::optional<capro::IdString_t>& instance,
                 const cxx::optional<capro::IdString_t>& event) noexcept;
 
-    /// @brief requests the serviceRegistryChangeCounter from the shared memory
+    /// @brief Searches all services that match the provided service description and applies a function to each of them
+    /// @param[in] service service string to search for, a nullopt corresponds to a wildcard
+    /// @param[in] instance instance string to search for, a nullopt corresponds to a wildcard
+    /// @param[in] event event string to search for, a nullopt corresponds to a wildcard
+    /// @param[in] function to apply to all matching services
+    void findService(const cxx::optional<capro::IdString_t>& service,
+                     const cxx::optional<capro::IdString_t>& instance,
+                     const cxx::optional<capro::IdString_t>& event,
+                     const cxx::function_ref<void(const ServiceContainer&)>& function) noexcept;
```

Comment:
```suggestion
                     const cxx::function_ref<void(const ServiceContainer&)>& callable) noexcept;
```

Reply:
Either is fine IMO and I prefer function as it is called functional programming for a reason.